src/main/java/com/esri/core/geometry/ogc/OGCGeometry.java

```diff
@@ -295,13 +303,14 @@ public boolean crosses(OGCGeometry another) {
 	}
 
 	public boolean within(OGCGeometry another) {
-		com.esri.core.geometry.Geometry geom1 = getEsriGeometry();
-		com.esri.core.geometry.Geometry geom2 = another.getEsriGeometry();
-		return com.esri.core.geometry.GeometryEngine.within(geom1, geom2,
-				getEsriSpatialReference());
+		return another.contains(this);
 	}
 
 	public boolean contains(OGCGeometry another) {
+		if (another.geometryType() == OGCConcreteGeometryCollection.TYPE) {
+			return another.contains(this);
```

Comment:
`this.contains(another)` is not the equal to `another.contains(this)`, it is?

The following test in [TestOGCContains.java.txt](https://github.com/Esri/geometry-api-java/files/2079969/TestOGCContains.java.txt) fails:

`assertContains("MULTIPOINT (1 2, 3 4)", "GEOMETRYCOLLECTION (POINT (1 2))");`

Reply:
@stolstov Sergey, thanks. How would you like us to use the fork?

Reply:
@mbasmanova I'd like to use the fork to finish this PR. Once done, I'll merge the forked branch here and then merge this PR. It would also make it simpler for you if you wish to add a commit.